From c18138c2435c50fcb04ae53c565229508fb5d8d7 Mon Sep 17 00:00:00 2001
From: Chen Gang G <gang.g.chen@intel.com>
Date: Mon, 21 Jun 2021 22:19:53 +0800
Subject: [PATCH] kernelflinger: Fix compile issues.

1. undefine __linux__ since kernerflinger doesn't dependence
on linux kernel
2. Disable __REMOVED_IN(23) to avoid borningssl build failure

Tracked-On: OAM-97093
Signed-off-by: Chen Gang G <gang.g.chen@intel.com>
---
 libsslsupport/Android.mk | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/libsslsupport/Android.mk b/libsslsupport/Android.mk
index f9a0ac8..19d59b6 100644
--- a/libsslsupport/Android.mk
+++ b/libsslsupport/Android.mk
@@ -15,6 +15,7 @@ LOCAL_STATIC_LIBRARIES := libgnuefi libefi
 #endif
 LOCAL_MODULE := libsslsupport
 LOCAL_CFLAGS += -Wno-error
+LOCAL_CFLAGS += -U__linux__
 include $(BUILD_EFI_STATIC_LIBRARY)
 
 ifeq ($(KERNELFLINGER_SSL_LIBRARY),)
@@ -125,6 +126,11 @@ LOCAL_CFLAGS_64 :=
 LOCAL_CFLAGS_x86 :=
 LOCAL_CFLAGS_x86_64 :=
 
+#Workaround to disable __REMOVED_IN(23) in bionic stdio.h to avoid borningssl build failure
+KF_SSL_PATH := $(KERNELFLINGER_SSLSUPPORT_PATH)/borningssl
+$(shell cp bionic/libc/include/stdio.h $(KF_SSL_PATH))
+$(shell sed -i 's/__sF\[\] __REMOVED_IN(23)/__sF\[\]/g' $(KF_SSL_PATH)/stdio.h)
+LOCAL_CFLAGS += -I$(KF_SSL_PATH)
 LOCAL_CFLAGS += -std=c99
 LOCAL_CFLAGS += -I$(LOCAL_PATH)/include
 LOCAL_CFLAGS += -DOPENSSL_NO_THREADS_CORRUPT_MEMORY_AND_LEAK_SECRETS_IF_THREADED
@@ -135,4 +141,5 @@ LOCAL_CFLAGS += -Ibionic/libc/kernel/uapi/asm-x86
 LOCAL_CFLAGS += -Ibionic/libc/kernel/android/uapi
 LOCAL_CFLAGS += -D_LIBCPP_BUILDING_LIBRARY
 LOCAL_CFLAGS += -DOPENSSL_NO_THREADS
+LOCAL_CFLAGS += -U__linux__
 include $(BUILD_EFI_STATIC_LIBRARY)
-- 
2.25.1

